packages/arb-token-bridge-ui/src/util/networks.ts

```diff
@@ -190,3 +206,116 @@ export function getNetworkName(chainId: number) {
       return 'Unknown'
   }
 }
+
+export function getNetworkLogo(chainId: number) {
+  switch (chainId) {
+    // L1 networks
+    case ChainId.Mainnet:
+      return EthereumLogo
+
+    case ChainId.Rinkeby:
+      return EthereumLogo
+
+    case ChainId.Goerli:
+      return EthereumLogo
+
+    // L2 networks
+    case ChainId.ArbitrumOne:
+      return ArbitrumOneLogo
+
+    case ChainId.ArbitrumNova:
+      return ArbitrumNovaLogo
+
+    case ChainId.ArbitrumRinkeby:
+      return ArbitrumOneLogo
+
+    case ChainId.ArbitrumGoerli:
+      return ArbitrumOneLogo
+
+    default:
+      return EthereumLogo
+  }
+}
+
+type ExtendedWeb3Provider = Web3Provider & {
+  isMetaMask?: boolean
+  isImToken?: boolean
+}
+
+export const getSupportedNetworks = (chainId?: number) => {
```

Comment:
in all its occurrence, we are not passing in the chainId, is this arg added for future usage or is it meant to be used in this PR?

Reply:
Yeah, the abstraction for this function is weird. I'm more for just inlining the networks where needed, like:

```
<MainNetworkNotSupported
  supportedNetworks={
    isNetwork(currentChainId).isTestnet 
      ? [ChainId.Goerli, ChainId.ArbitrumGoerli]
      : [ChainId.Mainnet, ChainId.ArbitrumOne, ChainId.ArbitrumNova]
  }
/>
```

IMO this is much easier to reason about.